client/src/tutorials/ZeroCopy/pages/1.md

```diff
@@ -0,0 +1,179 @@
+
+```rs
+#[account(zero_copy)];
+```
+
+Then you can define the repr which definies how the data will be packed. By default repr[c] will be used, so the C serialization.
+This will by default break options and enums in your structs because the C serialization is different from the Borsh serialization.
+You can also use:
+
+```rs
+#[repr(C)];
+or;
+#[repr(packed)];
+```
+
+which should remove all the extra space that the C serialization adds.
+
+Here is a list of different repr types
+[Repr types](https://doc.rust-lang.org/nomicon/other-reprs.html)
+
+[Space needed for different data types](https://book.anchor-lang.com/anchor_references/space.html)
+
+Keep in mind that when you use different repr types the data may not deserialize as expected in the client because for example options and enums may be packed differently.
+
+Next you replace Account with AccountLoader and then in you anchor program you can access the data using .load_mut()?
+Like this you can interact with the data of the account using copy_from_slice or mem copy without loading the whole account into memory.
+
+```rs
+pub fn set_data(ctx: Context<SetData>, string_to_set: String, index: u64) -> Result<()> {
+    let text_to_add_to_the_account = str::from_utf8(string_to_set.as_bytes()).unwrap();
+    msg!(text_to_add_to_the_account);
+
+    // Since the account is bigger that the heap space as soon as we access the whole account we will get a out of memory error
+    // let string = &ctx.accounts.data_holder.load_mut()?.long_string;
+    // let complete_string = str::from_utf8(string).unwrap();
+    // msg!("DataLength: {}", string.len());
+    // msg!("CompleteString: {}", complete_string);
+
+    // So the solution is use copy_from_slice and mem copy when we want to access data in the big account
+    ctx.accounts
+        .data_holder
+        .load_mut()?
+        .long_string[((index) as usize)..((index +912) as usize)]
+        .copy_from_slice(string_to_set.as_bytes());
+
+    Ok(())
+}
+
+// This will initialize the PDA with the maximum possible size of 10 Kb
+#[derive(Accounts)]
+pub struct Initialize<'info> {
+    #[account(init, seeds = [b"data_holder_zero_copy_v0",
+    signer.key().as_ref()],
+    bump,
+    payer=signer,
+    space= 10 * 1024 as usize)]
+    pub data_holder: AccountLoader<'info, DataHolder>,
+    #[account(mut)]
+    pub signer: Signer<'info>,
+    #[account(address = system_program::ID)]
+    pub system_program: Program<'info, System>,
+}
+
+#[account(zero_copy)]
+#[repr(packed)]
+pub struct DataHolder {
+    // 40952 = 40960 - 8 (account desciminator)
+    pub long_string: [u8; 40952],
+}
+
+#[derive(Accounts)]
+#[instruction(len: u16)]
+pub struct IncreaseZeroCopy<'info> {
+    #[account(mut,
+        realloc = len as usize,
+        realloc::zero = true,
+        realloc::payer=signer)]
+    pub data_holder: AccountLoader<'info, DataHolder>,
+    #[account(mut)]
+    pub signer: Signer<'info>,
+    #[account(address = system_program::ID)]
```

Comment:
Specifying the type `Program<'info, System>` already does this check, so checking again is redundant.

```suggestion
```